core/src/main/java/com/linecorp/armeria/common/HttpRequest.java

```diff
@@ -586,6 +587,23 @@ default HttpRequest mapTrailers(Function<? super HttpHeaders, ? extends HttpHead
         return of(headers(), stream);
     }
 
+    /**
+     * Applies specified {@link Function} to the {@link RequestHeaders} of this {@link HttpRequest}.
+     *
+     * <p>For example:<pre>{@code
+     * HttpRequest request = HttpRequest.of(HttpMethod.GET, "/items");
+     * HttpRequest result = request.peekHeaders(headers -> {
+     *      assertThat(headers.method()).isEqualTo(HttpMethod.GET);
+     *      assertThat(headers.path()).isEqualTo("/items");
+     * });
+     * }</pre>
+     */
+    default HttpRequest peekHeaders(Consumer<? super RequestHeaders> function) {
+        requireNonNull(function, "function");
+        function.accept(headers());
+        return this;
+    }
+
```

Comment:
```java
// Another way(certain types)
res.peek(ResponseHeaders.class, headers -> { ... })
   .map(HttpData.class, data -> { ...});
```
Above way is also possible, but I think `peekHeaders` looks better!
Cause we already use `mapHeaders`, `mapData` ... operators 🤔

Reply:
We usually implement a primitive method in `StreamMessage` and add additional methods in terms of the methods in `StreamMessage`.
Could we implement `StreamMessage.peek()` first?

Reply:
Please remove this method because it doesn't do anything. 😅 
We can just get the `RequestHeaders` via `headers()` method.

Reply:
> Could we implement StreamMessage.peek() first?

I got it! I'll update this PR.

> We can just get the RequestHeaders via headers() method.

oh.. we can use `headers()` instead! I'll remove this method!